### PR TITLE
#7 Task. Stenko Yaroslav

### PR DESCRIPTION
**Pull request by StYaroslav**
https://travis-ci.com/github/StYaroslav/js-assignments/builds/179799046

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -47,7 +61,16 @@ function* get99BottlesOfBeer() {
...
-    throw new Error('Not implemented');
+    yield 0
+    let [first, second] = [1, 1]
+    yield first
+    yield second
+    let i = 0
+    while (i <= 1000) {
+        yield first + second
+        first = [first + second, second = first][0]
+        i++
```
Comment (vramaniuk) [inline task/07-yield-tasks.js]: It works only for our test cases.
Actually we don't need `i` counter

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -145,7 +203,14 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    let sourceGenerator = generator()
+
+    let managePromises = (result) => {
+        return result.done ?  Promise.resolve(result.value) : Promise.resolve(result.value).then((result) =>
```
Comment (vramaniuk) [inline task/07-yield-tasks.js]: Please, avoid this code style, if-else is more readable than multiple conditional operators ? :

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by StYaroslav** [comment]
https://travis-ci.com/github/StYaroslav/js-assignments/builds/180652621

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**